var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -13,15 +13,14 @@ class NetcdfC(AutotoolsPackage):
 
     homepage = "http://www.unidata.ucar.edu/software/netcdf"
     git      = "https://github.com/Unidata/netcdf-c.git"
-    url      = "ftp://ftp.unidata.ucar.edu/pub/netcdf/netcdf-c-4.7.3.tar.gz"
+    url      = "https://www.unidata.ucar.edu/downloads/netcdf/ftp/netcdf-c-4.7.4.tar.gz"
 
     def url_for_version(self, version):
+        url = ['https://www.unidata.ucar.edu/downloads/netcdf/ftp/netcdf']
         if version >= Version('4.6.2'):
-            url = "ftp://ftp.unidata.ucar.edu/pub/netcdf/netcdf-c-{0}.tar.gz"
-        else:
-            url = "ftp://ftp.unidata.ucar.edu/pub/netcdf/netcdf-{0}.tar.gz"
-
-        return url.format(version.dotted)
+            url.append("-c")
+        url.extend(["-", str(version.dotted), ".tar.gz"])
+        return "".join(url)
```

Comment:
Do you want to compete? Let's compete :)
```python
@when('@:4.6.1')
def url_for_version(self, version):
    prefix, _, suffix = self.url.rpartition('netcdf-c')
    return spack.url.substitute_version(
        '{0}netcdf{1}'.format(prefix, suffix), self.url_version(version))
```

Reply:
😆 I wasn't trying to compete -- just trying to make it easier to understand the difference between the version-dependent URLs. I thought that building up the final URL piece-by-piece made it clearest -- but I'm happy to change to whatever you suggest!